### DISCUSSION TITLE
non-asyncio?

### DISCUSSION THREAD

**Discussion by hemna** [python-telegram-bot/python-telegram-bot#4419] [Feature Suggestions]
Hello,
  I used to use this library a few years back when it wasn't asyncio.  Is there an option to use this library without asyncio?  I have a heavily threaded application and isn't compatible with asyncio.

**Comment by Bibo-Joshi** [comment]
Hi. PTB does not provide a synchronous interface anymore, no. This is decision was made and announced back in 2021-01 https://t.me/pythontelegrambotchannel/94. I realize that this means that updating you threaded application will be rather painful, but adding a synchronous interface is not on any roadmap

**Reply by hemna** [reply]
ok that's too bad.  That makes this library useless for my app then.  'updating' a threaded app is not updating, it requires a complete rewrite to do asyncio properly.   I can't imagine why this library switched and thought it wouldn't make it impossible for threaded apps to use it anymore.   

thanks for the info, but this library is now dead.

**Reply by Poolitzer** [reply]
> I can't imagine why this library switched 

It is a request based library. Requests work best in an asyncio environment.

**Reply by Bibo-Joshi** [reply]
Sorry to hear that PTB does not fit your use case anymore. I trust that you can understand that we have to make decisions based on what is useful for the majority of use cases and also based on the limited capacity that the development team can put into this library in their spare time after work.
